snap/validate.go

```diff
@@ -602,6 +616,19 @@ func ValidateApp(app *AppInfo) error {
 		return fmt.Errorf(`"daemon" field contains invalid value %q`, app.Daemon)
 	}
 
+	switch app.DaemonMode {
+	case "":
+		if app.Daemon != "" {
+			return fmt.Errorf(`"daemon-mode" should be set for daemons`)
```

Comment:
in practice we should not get here because of the default,  anyway  /should/must/